drivers/advanced_io/advanced_io.c

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (C) 2017 Philipp-Alexander Blum <philipp-blum@jakiku.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+#include "periph/gpio.h"
+
+typedef enum {
+    MSBFIRST,
+    LSBFIRST
+} advanced_io_bit_order_t;
+
+uint8_t advanced_io_shift_in(gpio_t data_pin, gpio_t clock_pin, advanced_io_bit_order_t bit_order){
+    uint8_t byte = 0x00;
```

Comment:
IMO the byte should be reversed once at the end, using a factored-out function. that way it can more easily be factored into another place to be shared (e.g., bitarithm.h, with a way for configuring the best algorithm(-tradeoff), but that can be done later) .

At that point, isn't it more natural to call to call ```bit_reverse(advanced_io_shift_in(...))``` than ```advanced_io_shift_in(..., MSBFIRST)```?

Reply:
It's the way how the Arduino lib is doing it. But yes would be also fine. Reusing functions makes totally sense. How about using ```bit_reverse(advanced_io_shift_in(...))``` within a wrapper function?

Reply:
@kaspar030 Done. Removed the the enum.